From f3994a18e5e797b160514147820a245f795afb0d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:29 -0400
Subject: [usb] sierra: clean up suspend

Message-id: <1410887463-20674-438-git-send-email-dzickus@redhat.com>
Patchwork-id: 95607
O-Subject: [RHEL7 PATCH 437/591] USB: sierra: clean up suspend
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 7d8825bed46a28688f97cc934ecb326cc4ce2d2e
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:06 2014 +0200

    USB: sierra: clean up suspend

    Clean up suspend() somewhat and make sure to always set the suspended
    flag (although it's only used for runtime PM) in order to match
    resume().

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index 4b6d0ff..1d42e83 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -964,22 +964,18 @@ static void stop_read_write_urbs(struct usb_serial *serial)
 
 static int sierra_suspend(struct usb_serial *serial, pm_message_t message)
 {
- struct sierra_intf_private *intfdata;
- int b;
+ struct sierra_intf_private *intfdata = usb_get_serial_data(serial);
 
+ spin_lock_irq(&intfdata->susp_lock);
  if (PMSG_IS_AUTO(message)) {
-  intfdata = usb_get_serial_data(serial);
-  spin_lock_irq(&intfdata->susp_lock);
-  b = intfdata->in_flight;
-
-  if (b) {
+  if (intfdata->in_flight) {
    spin_unlock_irq(&intfdata->susp_lock);
    return -EBUSY;
-  } else {
-   intfdata->suspended = 1;
-   spin_unlock_irq(&intfdata->susp_lock);
   }
  }
+ intfdata->suspended = 1;
+ spin_unlock_irq(&intfdata->susp_lock);
+
  stop_read_write_urbs(serial);
 
  return 0;
-- 
1.7.1